### PR TITLE
Modify props

### PR DESCRIPTION
**Pull request by no-yan**
Based on https://github.com/Monchi/eslint-plugin-chakra-ui/pull/13#pullrequestreview-866960683

### CODE REVIEW

**Comment by no-yan** [comment]
I must consider pros and cons of rearrangement.
- nice and meaningful
- hard to automation

File: src/lib/getPriorityIndex.ts [hunk 1]
```diff
@@ -35,7 +61,7 @@ const priorityGroups: readonly PriorityGroup[] = [
       "fontFamily",
       "fontSize",
       "fontWeight",
-      "fontHeight",
+      "lineHeight",
```
Comment (Monchi) [inline src/lib/getPriorityIndex.ts:64]: :+1:

File: src/lib/getPriorityIndex.ts [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
-      name: string;
-      keys?: undefined;
-      includes: (key: string) => boolean;
-    };
-
-const directions = ["Top", "Right", "End", "Bottom", "Left", "Start"];
+type PriorityGroup = {
+  name: string;
+  keys: readonly string[];
+  isIncluded?: undefined;
```
Comment (Monchi) [inline src/lib/getPriorityIndex.ts:4]:
```suggestion
```

**Review by Monchi — COMMENTED** [review]
This simple approach seems good for now.

Perhaps we can automatically generate those rules from the definition in `@chakra-ui/styled-system`. We need to investigate though.
ref: https://github.com/chakra-ui/chakra-ui/blob/97fc1c4effa31482bafdd1066a8652611c15b4bf/packages/styled-system/src/config/space.ts

**Comment by no-yan** [comment]
In this approach,  I noticed some ugly points regarding rtl-related props.

- example : `marginStart` comes after `marginEnd`

Maybe we should change that in the future.

> Perhaps we can automatically generate those rules from the definition in @chakra-ui/styled-system. We need to investigate though.

I think so too! Let's automate it if we can.

**Comment by Monchi** [comment]
@no-yan Is this ready for release?

**Comment by no-yan** [comment]
Yes, definitely!

**Comment by no-yan** [comment]
When I expect improvements in the review, it will be easier to see when to merge if I open PR as a draft.
I'll do that next time!

**Pull request merged**